packages/core/src/Credentials.ts

```diff
@@ -24,30 +22,14 @@ export class Credentials extends ICredentials {
 	/**
 	 * Sets new credential object
 	 */
-	setData(data: ICredentialDataDecryptedObject, encryptionKey: string): void {
-		this.data = AES.encrypt(JSON.stringify(data), encryptionKey).toString();
-	}
-
-	/**
-	 * Sets new credentials for given key
-	 */
-	setDataKey(key: string, data: CredentialInformation, encryptionKey: string): void {
-		let fullData;
-		try {
-			fullData = this.getData(encryptionKey);
-		} catch (e) {
-			fullData = {};
-		}
-
-		fullData[key] = data;
-
-		return this.setData(fullData, encryptionKey);
+	setData(data: ICredentialDataDecryptedObject): void {
+		this.data = this.cipher.encrypt(JSON.stringify(data));
```

Comment:
Stringifying an object for encryption happens frequently. Perhaps `encrypt` should be able to stringify when needed.